programs/mpl-core/src/plugins/external_plugins.rs

```diff
@@ -0,0 +1,144 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use solana_program::pubkey::Pubkey;
+
+use strum::EnumCount;
+
+use super::{Authority, DataStore, ExternalCheckResult, LifecycleEvent, LifecycleHook, Oracle};
+
+/// List of third party plugin types.
+#[repr(C)]
+#[derive(
+    Clone, Copy, Debug, BorshSerialize, BorshDeserialize, Eq, PartialEq, EnumCount, PartialOrd, Ord,
+)]
+pub enum ExternalPluginKey {
+    /// Lifecycle Hook.  Extra accounts are specified in the external plugin header.  The attached
+    /// `Pubkey` is the hooked program, which is called at specified lifecycle events and will
+    /// return a validation result and new data to store.
+    LifecycleHook(Pubkey),
+    /// Oracle.  Get a `ValidationResult` result from an account either specified by or derived
+    /// from the attached `Pubkey`.
+    Oracle(Pubkey),
+    /// Data Store.  Arbitrary data that can be written to by the attached `Authority`.  Note this
+    /// is different then the plugin authority as it cannot update/revoke authority for the plugin.
+    DataStore(Authority),
```

Comment:
Yes because it is the key it is the unique Plugin indicator, so changing that should require a new plugin.

Originally I thought of it as DataStore(Pubkey), but thought it would be more flexible to allow UpdateAuthority and Owner to be able to write to that data.  None I guess would mean its an immutable blob of 3P data.

Reply:
I think None and Owner make sense if someone wants immutable data or to give the owner something to write to.

Reply:
Resolved offline.